packages/client/hooks/useBeforeUnload.ts

```diff
@@ -0,0 +1,13 @@
+import {useRef, useEffect} from 'react'
+
+export const useBeforeUnload = (callback: () => any) => {
```

Comment:
oooo i like where this is headed!
this solution works right now, but what if in the future discussionId changes from "foo" to "bar"? the callback is persisted & points to the old value of discussionId. That'd be a tricky bug to track down.

easy fix:

`const cb = useEventCallback(callback)`

then you can pass cb to window event listeners

Reply:
oh, that's 100% right 🤦